crates/bevy_ecs/src/world/mod.rs

```diff
@@ -444,6 +444,29 @@ impl World {
     /// assert_eq!(world.get::<Position>(entities[0]).unwrap(), &Position { x: 1.0, y: 0.0 });
     /// assert_eq!(world.get::<Position>(entities[1]).unwrap(), &Position { x: 0.0, y: 1.0 });
     /// ```
+    ///
+    /// To iterate over entities in a deterministic order,
+    /// sort the results of the query using the desired component as a key.
+    ///
+    /// ```
+    /// use bevy_ecs::{entity::Entity, world::World};
+    /// let mut world = World::new();
+    /// let a = world.spawn().insert_bundle((2, "abc")).id();
+    /// let b = world.spawn().insert_bundle((3, "xyz")).id();
+    /// let c = world.spawn().insert_bundle((1, "def")).id();
+    /// let mut entities = world.query::<(Entity, &i32, &str)>()
+    ///     .map(|(e, &i &s)| (e, i, &s)) // Copy out of the world
+    ///     .collect::<Vec<_>>();
+    /// // Sort by `i32` component
+    /// entities.sort_by(|x, y| x.1.cmp(&y.1));
+    /// for (index, entity) in entities.iter().enumerate() {
+    ///     match index {
+    ///         0 => assert!(entity == (c, 1, "def")),
+    ///         1 => assert!(entity == (a, 2, "abc")),
+    ///         2 => assert!(entity == (b, 3, "xyz")),
+    ///     }
+    /// }
+    /// ```
```

Comment:
```suggestion
    /// assert_eq!(entity[0], (c, 1, "def"));
    /// assert_eq!(entity[1], (c, 2, "abc"));
    /// assert_eq!(entity[2], (c, 3, "xyz"));
    /// ```
```

Reply:
It was before, but @alice-i-cecile asked for a loop.

Reply:
Oh well I'll have to disagree with @alice-i-cecile here, I feel the loop with an index to assert is very contrived.

Could you still replace the `assert!` by `assert_eq!`?

Reply:
> I feel the loop with an index to assert is very contrived.

It absolutely is, but I wanted an example to clearly demonstrate to beginners how you'd finish the pattern in an actual game. Working with non-trivial loops, mapping and so on is harder than you'd expect if you're new to programming.

Reply:
@mockersf, leaving it as is with addressing the comment about assertions, right?

Reply:
Ok for me